dev/sg/internal/images/images.go

```diff
@@ -13,15 +16,31 @@ import (
 	"github.com/opencontainers/go-digest"
 	"sigs.k8s.io/kustomize/kyaml/kio"
 	"sigs.k8s.io/kustomize/kyaml/yaml"
+	k8syaml "sigs.k8s.io/yaml"
 
 	"github.com/sourcegraph/sourcegraph/dev/sg/internal/stdout"
 	"github.com/sourcegraph/sourcegraph/lib/errors"
 )
 
 var seenImageRepos = map[string]imageRepository{}
 
-func Parse(path string, creds credentials.Credentials) error {
+type DeploymentType string
 
+const (
+	DeploymentTypePlain DeploymentType = "plain"
```

Comment:
I don't think we need to worry about backcompat here, and `plain` is very deceptive because from what I understand this is the opposite of plain, it is k8s-only, so I think we should take the opportunity to make this more descriptive:

```suggestion
	DeploymentTypeKubernetes DeploymentType = "k8s"
```

FWIW "plain" would be a simple regex replace over all content IMO